index.js

```diff
@@ -85,109 +81,138 @@ PersonaClient.prototype.validateToken = function (token, scope, overridingScope,
+            if(error) {
+                return next(ERROR_TYPES.VALIDATION_FAILURE, null);
+            }
+
+            if(scope && decodedToken.hasOwnProperty("scopeCount")) {
+                // Too many scopes to put in the payload so we'll have to ask Persona
+                return headScopeThenVerify(scope, next);
+            } else if (!scope || _.includes(decodedToken.scopes, "su") || _.includes(decodedToken.scopes, scope)) {
+                return next(null, "ok");
+            } else {
+                return next(ERROR_TYPES.INSUFFICIENT_SCOPE, null);
+            }
+        });
+    };
+    verifyToken = verifyToken.bind(this);
+
+    if (this.publicKey == null) {
+        var options = {
+            hostname: this.config.persona_host,
+            port: this.config.persona_port,
+            path: "/oauth/keys",
+            method: "GET"
+        };
+
+        this.http.request(options, function onSuccess(response) {
+            if(response.statusCode === 200) {
+                var publicKey = "";
+                response.on("data", function onData(chunk) {
+                    publicKey += chunk;
+                });
+                response.on("end", function onEnd() {
+                    return verifyToken(publicKey);
+                });
+            } else {
+                return next(ERROR_TYPES.COMMUNICATION_ISSUE, null);
+            }
+        }).on("error", function onError(error) {
+            return next(error, null);
+        }).end();
+    } else {
+        return verifyToken(this.publicKey);
+    }
 };
 
 /**
- * Express middleware that can be used to verify a token
- * @param req
- * @param res
- * @param next
+ * Express middleware that can be used to verify a token.
+ * @param {Object} request - HTTP request object.
+ * @param {Object} response - HTTP response object.
+ * @callback next - Called with either an error as the first param or "ok" as the result in the second param.
+ * @param {string} scope - @Deprecated since version 1.3.0 set the required scope as a request param. (su scope will be automatically checked for).
  */
-PersonaClient.prototype.validateHTTPBearerToken = function (req, res, next, scope) {
-    var token = this.getToken(req);
-    this.validateToken(token, req.param("scope"), scope, function (err, validatedBy) {
-        if (!err) {
-            next(null, validatedBy);
+PersonaClient.prototype.validateHTTPBearerToken = function (request, response, next, scope) {
+    var token = this.getToken(request);
+    this.validateToken(token, request.param("scope"), scope, function (error, validationResult) {
+        if (!error) {
+            next(null, validationResult);
             return;
         }
 
-        switch(err) {
+        switch(error) {
         case ERROR_TYPES.INVALID_TOKEN:
-            res.status(401);
-            res.json({
+            response.status(401);
+            response.json({
```

Comment:
I believe `response.json` has the signature of `response.json(status, body)`. The two method calls above can be done in one.